StepsToCare/CareEntry.ascx.cs

```diff
@@ -432,35 +451,45 @@ protected void lbSave_Click( object sender, EventArgs e )
                     person = new PersonService( rockContext ).Get( ppPerson.PersonId.Value );
                 }
 
+                string originalSubmitter = History.GetValue<PersonAlias>( careNeed.SubmitterPersonAlias, careNeed.SubmitterAliasId.ToStringSafe().AsIntegerOrNull(), rockContext );
+                string submitter = History.GetValue<PersonAlias>( null, ppSubmitter.PersonAliasId, rockContext );
+                History.EvaluateChange( changes, "Submitter", originalSubmitter, submitter );
                 careNeed.SubmitterAliasId = ppSubmitter.PersonAliasId;
 
                 if ( careNeed.StatusValueId != dvpStatus.SelectedDefinedValueId && dvpStatus.SelectedDefinedValueId == snoozedValueId )
                 {
+                    History.EvaluateChange( changes, "Snooze Date/Time", careNeed.SnoozeDate, RockDateTime.Now );
                     careNeed.SnoozeDate = RockDateTime.Now;
                 }
 
+                History.EvaluateChange( changes, "Status", DefinedValueCache.GetName( careNeed.StatusValueId ), DefinedValueCache.GetName( dvpStatus.SelectedDefinedValueId ) );
                 careNeed.StatusValueId = dvpStatus.SelectedDefinedValueId;
 
-                careNeed.CategoryValueId = dvpCategory.SelectedValue.AsIntegerOrNull();
+                History.EvaluateChange( changes, "Category", DefinedValueCache.GetName( careNeed.CategoryValueId ), DefinedValueCache.GetName( dvpCategory.SelectedDefinedValueId ) );
+                careNeed.CategoryValueId = dvpCategory.SelectedDefinedValueId;
 
-                if ( dpDate.SelectedDateTime.HasValue )
-                {
-                    careNeed.DateEntered = dpDate.SelectedDateTime.Value;
-                }
+                DateTime? dateEnteredDateTime = dpDate.SelectedDateTimeIsBlank ? null : dpDate.SelectedDateTime;
+                History.EvaluateChange( changes, "Date Entered", careNeed.DateEntered, dateEnteredDateTime );
+                careNeed.DateEntered = dpDate.SelectedDateTime;
 
                 if ( careNeed.DateEntered.HasValue )
                 {
                     dateDifference = ( careNeed.DateEntered.Value - DateTime.Now ).TotalDays;
                 }
 
+                History.EvaluateChange( changes, "Workers Only", careNeed.WorkersOnly, cbWorkersOnly.Checked );
                 careNeed.WorkersOnly = cbWorkersOnly.Checked;
 
+                History.EvaluateChange( changes, "Custom Follow Up", careNeed.CustomFollowUp, cbCustomFollowUp.Checked );
                 careNeed.CustomFollowUp = cbCustomFollowUp.Checked;
+                History.EvaluateChange( changes, "Follow Up After", careNeed.RenewPeriodDays, numbRepeatDays.IntegerValue );
```

Comment:
Anywhere RenewPeriodDays is used, the label should be "Follow Up After" not Renew Period Days.